js/ui/appIconBar.js

```diff
@@ -631,21 +641,52 @@ const ScrolledIconList = GObject.registerClass({
 
         appSys.connect('installed-changed', this._onInstalledChanged.bind(this));
         appSys.connect('app-state-changed', this._onAppStateChanged.bind(this));
+
+        this._parentalControlsManager.connect('changed', () => {
+            for (let [app, appButton] of this._taskbarApps) {
+                if (app.state == Shell.AppState.STOPPED &&
+                    !this._parentalControlsManager.shouldShowApp(app.get_app_info()))
```

Comment:
How does this interact with pinned buttons? Are they filtered as well?

Reply:
Yes,  actually pinned buttons are the only ones that should be filtered here as normal apps that have state `STOPPED` should not appear in `this._taskbarApps`. No need to check if the app is pinned cause if it happens (shouldn't) that an app is `STOPPED` and in `_taskbarApps` we want to hide it anyway.